src/main/webapp/assets/js/productpage.js

```diff
@@ -36,9 +51,9 @@ function populateImageGallery(photoUrlList) {
   const urlListLength = photoUrlList.length;
```

Comment:
If there are no images, the image gallery div will still take up space and there is a white empty space (can be seen in the product Some Images)

You can hide the space by setting the display to none

element.style.display = none

https://www.edureka.co/community/62212/explain-difference-between-visibility-hidden-display-none#:~:text=visibility%3Ahidden%2D%20It%20is%20not,hidden%20and%20takes%20no%20space.

Reply:
You should also hide the menu link and contact link (the hyperlinks) if you find out that they're null or empty

Reply:
done!